### PR TITLE
Add fix option

### LINKED ISSUE #46: Allow fixes to be applied to the output directory
If I implemented a feature to apply the ESLint fixes to the output directory, would that be accepted?  Just want to get some confirmation that that would be desirable before starting work on it.

Adding `fix: true` to the options passed on to the ESLint node API isn't quite enough; you also have to invoke [`outputFixes`](http://eslint.org/docs/developer-guide/nodejs-api#outputfixes).

### PR DESCRIPTION
**Pull request by alexlafroscia**
If `fix: true` is present in the ESLint options passed to the Broccoli plugin, then the output file will be the "fixed" version rather than the original.

Closes #46

### CODE REVIEW

**Comment by alexlafroscia** [comment]
I ended up increasing the timeout to 3 seconds to get the tests to pass locally.  In CI they seem to run under 2 seconds without issue, but locally the first test of each set was taking around 2.5 seconds.  I don't think it really matters, but if the increase bugs anyone I'm happy to put it back, since CI will likely pass either way.

It would be really nice to get to the bottom of why that first test takes so much longer than the others in both the `main` and `format` sets of tests.

File: lib/index.js [hunk 1]
```diff
@@ -169,8 +173,15 @@ EslintValidationFilter.prototype.cacheKeyProcessString = function cacheKeyProces
 
 EslintValidationFilter.prototype.processString = function processString(content, relativePath) {
   // verify file content
-  const configPath = path.join(this.eslintrc, relativePath);
-  const output = this.cli.executeOnText(content, configPath);
+  const filePath = path.join(this.eslintrc, relativePath);
```
Comment (BrianSipple) [inline lib/index.js]: I noticed this recently, too. 👍

File: lib/index.js [hunk 2]
```diff
@@ -169,8 +173,15 @@ EslintValidationFilter.prototype.cacheKeyProcessString = function cacheKeyProces
...
+  const output = this.cli.executeOnText(content, filePath);
+  const { results } = output;
+  const [ firstResult ] = results;
+
+  if (this.applyFixes && firstResult.output) {
+    content = firstResult.output;
```
Comment (BrianSipple) [inline lib/index.js]: Shouldn't this reassignment fail after initializing `output` with `const`?

**Comment by BrianSipple** [comment]
I'd be curious to see the diffs after rebasing, but overall, I think this is on a good track.

**Pull request closed without merging**